common/lib/xmodule/xmodule/x_module.py

```diff
@@ -398,7 +398,7 @@ def get_content_titles(self):
         the display_names of all its leaf nodes.
         """
         if self.has_children:
-            return sum((child.get_content_titles() for child in self.get_children()), [])
+            return sum((child.get_content_titles() for child in self.get_display_items()), [])
```

Comment:
If `get_content_titles` calls `get_children` then the `get_children` in `XModuleMixin` gets called, which doesn't have access check. [get_children](https://github.com/edx/edx-platform/blob/ammar/tnl-1035/common/lib/xmodule/xmodule/x_module.py#L405) in `XModuleMixin`
If `get_content_titles` calls `get_display_items` then the `get_children` in `XModule` gets called, which does have access check. [get_children](https://github.com/edx/edx-platform/blob/ammar/tnl-1035/common/lib/xmodule/xmodule/x_module.py#L658) in `XModule`